File: backend/policy-service/src/main/java/com/insurance/policyservice/model/Claim.java
package com.insurance.policyservice.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "claims")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Claim {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private LocalDate date;

    @NotBlank
    @Column(length = 500)
    private String description;

    @NotNull
    @Positive
    private BigDecimal claimedAmount;

    private BigDecimal settledAmount = BigDecimal.ZERO;;

    @Enumerated(EnumType.STRING)
    private ClaimStatus status = ClaimStatus.PENDING;

    @NotNull
    private Long policyId;
}

File: frontend/src/types/Claim.ts
export enum ClaimStatus {
  PENDING = "PENDING",
  APPROVED = "APPROVED",
  REJECTED = "REJECTED",
  SETTLED = "SETTLED",
}

export interface Claim {
  id?: number
  date: string
  description: string
  claimedAmount: number
  settledAmount?: number
  status: ClaimStatus
  policyId: number
}

